extension/lib/mergeCartsOnLogin.js

```diff
@@ -10,7 +10,15 @@ const ShopgateCartPipeline = require('./shopgate/CartExtensionPipeline')
 module.exports = async (context, input) => {
   try {
     const anonymousPipeline = ShopgateCartPipeline.createForDevice(context)
-    const loggedInPipeline = ShopgateCartPipeline.create(context)
+    const loggedInPipeline = ShopgateCartPipeline.createForUser(context)
+
+    const anonymousCartId = await anonymousPipeline.getCartId()
+    const loggedInUserCartId = await loggedInPipeline.getCartId()
+    // if there is no user cart id available, we can just reuse the one we have from the device and we're done.
+    if (!loggedInUserCartId) {
+      await loggedInPipeline.setCartId(anonymousCartId)
```

Comment:
Are we able to assign this anonymous cart in BigCommerce to the user? I thought Christian once mentioned that its maybe possible.

Reply:
He did mention it, but didn't implement it this way. I had the same thought at the time but for us it was enough to just move the cart from one storage to another.
I'd agree that this would be the cleanest way - there is a new parameter https://developer.bigcommerce.com/api/v3/cart.html?json-doc#put-carts-cartid 
and we could use the call to set the customer id.